packages/flutter_tools/lib/src/android/android_workflow.dart

```diff
@@ -42,63 +106,56 @@ class AndroidWorkflow extends DoctorValidator implements Workflow {
-        final String androidHomeDir = platform.environment[kAndroidHome];
-        messages.add(new ValidationMessage('$kAndroidHome = $androidHomeDir'));
-      }
+    if (androidSdk.latestVersion != null) {
+      sdkVersionText = 'Android SDK ${androidSdk.latestVersion.buildToolsVersionName}';
 
-      final List<String> validationResult = androidSdk.validateSdkWellFormed();
+      messages.add(new ValidationMessage(
+        'Platform ${androidSdk.latestVersion.platformVersionName}, '
+        'build-tools ${androidSdk.latestVersion.buildToolsVersionName}'
+      ));
+    }
 
-      if (validationResult.isEmpty) {
-        // Empty result means SDK is well formed.
-        // The SDK also requires a valid Java JDK installation.
-        const String _kJdkDownload = 'https://www.oracle.com/technetwork/java/javase/downloads/';
-        String javaVersion;
+    if (platform.environment.containsKey(kAndroidHome)) {
+      final String androidHomeDir = platform.environment[kAndroidHome];
+      messages.add(new ValidationMessage('$kAndroidHome = $androidHomeDir'));
+    }
 
-        try {
-          printTrace('java -version');
+    final List<String> validationResult = androidSdk.validateSdkWellFormed();
 
-          final ProcessResult result = processManager.runSync(<String>['java', '-version']);
-          if (result.exitCode == 0) {
-            javaVersion = result.stderr;
-            final List<String> versionLines = javaVersion.split('\n');
-            javaVersion = versionLines.length >= 2 ? versionLines[1] : versionLines[0];
-           }
-        } catch (error) {
-        }
-
-        if (javaVersion == null) {
-          messages.add(new ValidationMessage.error(
-              'No Java Development Kit (JDK) found; you can download the JDK from $_kJdkDownload.'
-          ));
-        } else {
-          messages.add(new ValidationMessage(javaVersion));
-          type = ValidationType.installed;
-        }
-      } else {
-        messages.addAll(validationResult.map((String message) {
-          return new ValidationMessage.error(message);
-        }));
-        messages.add(new ValidationMessage(
+    if (validationResult.isNotEmpty) {
+      // Android SDK is not functional.
+      messages.addAll(validationResult.map((String message) {
+        return new ValidationMessage.error(message);
+      }));
+      messages.add(new ValidationMessage(
           'Try re-installing or updating your Android SDK,\n'
           'visit https://flutter.io/setup/#android-setup for detailed instructions.'));
-      }
+      return new ValidationResult(ValidationType.partial, messages, statusInfo: sdkVersionText);
+    }
+
+    // Now check for the JDK.
+    final String javaBinary = _findJavaBinary();
+    if (javaBinary == null) {
+      messages.add(new ValidationMessage.error(
+          'No Java Development Kit (JDK) found; You must have the environment '
+          'variable JAVA_HOME set and the java binary in your PATH. '
```

Comment:
@goderbauer You don't strictly need both but if a user doesn't have a working Java environment I think it's better to give them more direction here.